### PR TITLE
Show password config on validation failure on taping

### PR DESCRIPTION
**Pull request by bdjelaili**
closes gravitee-io/issues#5128

### CODE REVIEW

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
     String ERROR_DESCRIPTION_PARAM_KEY = "error_description";
     String SUCCESS_PARAM_KEY = "success";
     String WARNING_PARAM_KEY = "warning";
+    String PASSWORD_CONFIG_PARAM_KEY = "passwordConfig";
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java:44]: Replace passwordConfig for passwordSettings (to keep consistency with the backend).

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/registration.html [hunk 2]
```diff
@@ -57,6 +59,28 @@
...
+
+                            <p th:if="${passwordConfig.passwordInclude.toString()=='NUMBERS_AND_SPECIAL_CHARACTERS'}"
+                               id="specialChar" class="invalid">Contains number and special character</p>
+                        </div>
+                        <p th:if="${passwordConfig.lettersInMixedCase!= null && passwordConfig.lettersInMixedCase}"
+                           id="mixedCase" class="invalid">Contains letters in mixed case</p>
+                        <p th:if="${passwordConfig.maxConsecutiveLetters!= null}"
+                           id="maxConsecutiveLetters" class="valid">Max <span th:text="${passwordConfig.maxConsecutiveLetters}"/> consecutive letters or numbers
+                        </p>
+                    </div>
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/registration.html:82]: Move this block under the password input.

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/reset_password.html [hunk 3]
```diff
@@ -25,22 +26,47 @@
...
+                    <p th:if="${passwordConfig.passwordInclude.toString()=='NUMBERS_AND_SPECIAL_CHARACTERS'}"
+                       id="specialChar" class="invalid">Contains number and special character</p>
+                </div>
+                <p th:if="${passwordConfig.lettersInMixedCase!= null && passwordConfig.lettersInMixedCase}"
+                   id="mixedCase" class="invalid">Contains letters in mixed case</p>
+                <p th:if="${passwordConfig.maxConsecutiveLetters!= null}"
+                   id="maxConsecutiveLetters" class="valid">Max <span
+                        th:text="${passwordConfig.maxConsecutiveLetters}"/> consecutive letters or numbers
+                </p>
             </div>
```
Comment (tcompiegne) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/reset_password.html:51]: Move this block under the password input.

**Review by tcompiegne — CHANGES_REQUESTED** [review]

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java (hunk 1 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ConstantKeys.java:44]: OK

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/registration.html (hunk 2 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/registration.html:82]: ok

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/reset_password.html (hunk 3 above)
Comment (bdjelaili) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/reset_password.html:51]: ok

**Pull request merged**